src/remoteCollection/collect.js

```diff
@@ -86,7 +86,8 @@ function prepareHeaders(headers, ctx) {
  */
 function sendRemoteRequest(generator, connection, simpleOauth=null) {
   return new Promise((resolve) => {
-    const remoteUrl = prepareUrl(generator);
+    // Add the url to the generator so the handler has access to it later.
+    generator.url = prepareUrl(generator);
```

Comment:
adding the preparedUrl to just generator.url does not add any context. Probably adding it to generator.generatorTemplate.connection.url will be more helpful , considering that the connection object of generatorTemplate is for this purpose.

Reply:
I think doing it this way makes it clear that this is the prepared url and not the unexpanded url template from generatorTemplate.connecton.url. Maybe it would be better to call it preparedUrl or requestUrl?
Also it's accessed once here and several times in handleCollectResponse so having it at the top level makes the code less verbose.

Reply:
Ok makes sense. Calling it preparedUrl probably would add more context than just calling it url.

Reply:
ok. changed to preparedUrl.